ui.apps/src/main/content/jcr_root/apps/core/fd/components/clientlib-core-forms-components-runtime/js/main.js

```diff
@@ -0,0 +1 @@
ength>=e,value:t}),maxItems:(e,t)=>({valid:t instanceof Array&&t.length<=e,value:t}),uniqueItems:(e,t)=>({valid:!e||t instanceof Array&&t.length===new Set(t).size,value:t}),minLength:(e,n)=>Object.assign(Object.assign({},t.Constraints.minimum(e,"string"==typeof n?n.length:0)),{value:n}),maxLength:(e,n)=>Object.assign(Object.assign({},t.Constraints.maximum(e,"string"==typeof n?n.length:0)),{value:n}),pattern:(e,t)=>{let n;return n="string"==typeof e?new RegExp(e):e,{valid:n.test(t),value:t}},required:(e,t)=>({valid:!e||null!=t&&""!==t,value:t}),enum:(e,t)=>({valid:e.indexOf(t)>-1,value:t})}}},t={};function n(r){var i=t[r];if(void 0!==i)return i.exports;var s=t[r]={exports:{}};return e[r].call(s.exports,s,s.exports,n),s.exports}n.d=(e,t)=>{for(var r in t)n.o(t,r)&&!n.o(e,r)&&Object.defineProperty(e,r,{enumerable:!0,get:t[r]})},n.o=(e,t)=>Object.prototype.hasOwnProperty.call(e,t),n.r=e=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var r={};(()=>{"use strict";function e(e,t){const n=e.dataset;let r=[],i=t;i=i.charAt(0).toUpperCase()+i.slice(1);const s=["is","hook"+i];for(let e in n)if(Object.prototype.hasOwnProperty.call(n,e)){let t=n[e];0===e.indexOf("cmp")&&(e=e.slice("cmp".length),e=e.charAt(0).toLowerCase()+e.substring(1),-1===s.indexOf(e)&&(r[e]=t))}return r}function t(e){let t=window.MutationObserver||window.WebKitMutationObserver||window.MozMutationObserver,n=document.querySelector("body");new t((function(t){t.forEach((function(t){var n=[].slice.call(t.addedNodes);n.length>0&&n.forEach((function(t){t.querySelectorAll&&[].slice.call(t.querySelectorAll(e.selectors.self)).forEach((function(t){let n=FormView.readData(t,e.IS).formcontainer,r=window.af.formsRuntime.view.formContainer[n];new e({element:t,formContainer:r})}))}))}))})).observe(n,{subtree:!0,childList:!0,characterData:!0})}n.r(r),n.d(r,{FormField:()=>s,createFormInstance:()=>i.createFormInstance,readData:()=>e,registerMutationObserver:()=>t});var i=n(289);class s{constructor(t){this.formContainer=t.formContainer,this.parent=t.parent||t.formContainer,this.element=t.element,this.options=e(this.element,this.getClass());let n=this.element.getElementsByTagName(this.getTagName());n&&n.length>0&&this.setId(n[0].id),this.bindEventListeners()}setId(e){this.id=e}getFormContainerPath(){return this.options.formcontainer}getId(){return this.id}bindEventListeners(){this.element.addEventListener("change",(e=>{this._model.value=e.target.value}))}setModel(e){this._model=e}getModel(){return this._model}getClass(){throw new Error("Not Implemented")}getTagName(){throw new Error("Not Implemented")}setValue(e){throw new Error("Not implemented")}subscribe(){this._model.subscribe((e=>{let t=e.target.getState();t.valid?this.setValue(t.value):(alert(t.errorMessage),this.setValue(null))}))}}window.af={formsRuntime:{model:{form:{}},view:{formContainer:{},utils:{}},event:{}}}})(),window.FormView=r})();
```

Comment:
should we check-in this file ? I think it is autogenerated, we can put it in .gitignore

Reply:
yes, its in TODOs. we should not checkin this file.